aQute.libg/test/aQute/lib/tag/TagTest.java

```diff
@@ -0,0 +1,289 @@
+package aQute.lib.tag;
+
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.ListIterator;
+import java.util.Map;
+
+import junit.framework.TestCase;
+
+public class TagTest extends TestCase {
+
+	public void testDtoToTag() {
+		Tag t;
+		ListIterator<Object> l;
+		Tag c;
+		Map<String,Object> map;
+		Map<String,Object> map2;
+
+		// Test Scalar
+		t = getTag(null);
+		assertTrue(t.content.size() == 0);
+
+		t = getTag("");
+		assertTrue(t.content.size() == 1);
+		assertTrue(((Tag) t.content.listIterator().next()).content.size() == 1);
```

Comment:
You may want to consider using [`assertj`](http://joel-costigliola.github.io/assertj/assertj-core.html) for these tests since it is easier to test things about lists like size, content, etc.